.github/workflows/ci.yml

```diff
@@ -61,6 +61,36 @@ jobs:
         env:
           RUSTDOCFLAGS: ${{ matrix.rust_channel == 'nightly' && '-Dwarnings --cfg=docsrs' || '-Dwarnings' }}
 
+  check-external-types:
+    name: Validate external types appearing in public API
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly-2023-10-10
+          # ^ sync with https://github.com/awslabs/cargo-check-external-types/blob/main/rust-toolchain.toml
+      - run: cargo install --locked cargo-check-external-types
```

Comment:
Do they publish binaries? If so, we should download them instead of building it for every PR/push to a PR.

Reply:
> Do they publish binaries? 

They do not.

Reply:
there is ways to cache artifacts on github CI. I think this is a suitable candidate.

Reply:
> there is ways to cache artifacts on github CI.

Can you point me towards what you mean? We're already including `~/.cargo/bin` in the [cache key](https://github.com/rustls/rcgen/blob/0318d2f0a4bb9e65c84d284acff8587bd418d837/.github/workflows/ci.yml#L78). I'm not sure if you had something else in mind.